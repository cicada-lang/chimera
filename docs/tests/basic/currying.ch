function f(x, y) {
  return [x, y]
}

print f
print f()
print f(1)
print f(1)(2)
print f(1, 2)

function g(x) {
  return (y) => [x, y]
}

print g
print g()
print g(1)
print g(1)(2)
print g(1, 2)
