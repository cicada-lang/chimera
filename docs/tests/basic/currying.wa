function f(x, y) {
  return [x, y]
}

print f()
print f(1)
print f(1)(2)
print f(1, 2)
