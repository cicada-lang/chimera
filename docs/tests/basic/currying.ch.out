(x, y) => {
  return [x, y]
}
$Curried((x, y) => {
  return [x, y]
}, 2, [])
$Curried((x, y) => {
  return [x, y]
}, 2, [1])
[1, 2]
[1, 2]
(x) => {
  return (y) => {
    return [x, y]
  }
}
$Curried((x) => {
  return (y) => {
    return [x, y]
  }
}, 1, [])
(y) => {
  return [x, y]
}
[1, 2]
[1, 2]
